src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @internal
+ */
+final class SocketCaster
+{
+    public static function castSocket(\Socket $socket, array $a, Stub $stub, bool $isNested): array
+    {
+        socket_getsockname($socket, $addr, $port);
+        $info = stream_get_meta_data(socket_export_stream($socket));
+
+        $uri = ($info['uri'] ?? '//');
+        if (str_starts_with($uri, 'unix://')) {
+            $uri .= $addr;
+        } else {
+            $uri .= \sprintf(str_contains($addr, ':') ? '[%s]:%s' : '%s:%s', $addr, $port);
+        }
+
+        $a += [
+            Caster::PREFIX_VIRTUAL.'uri' => $uri,
+            Caster::PREFIX_VIRTUAL.'timed_out' => $info['timed_out'],
+            Caster::PREFIX_VIRTUAL.'blocked' => $info['blocked'],
```

Comment:
atmark is now missing from my caster and we should be good

Reply:
Atmark added and test updated. PHP 8.2 doesn't fill the `uri` field with the protocol like it does in 8.3, so I removed the entry in this case.